src/marqo/tensor_search/tensor_search.py

```diff
@@ -345,10 +355,32 @@ def _add_documents_unstructured(config: Config, add_docs_params: AddDocsParams,
             # All the plain tensor/non-tensor fields are processed, now we process the multimodal fields
             if document_is_valid and add_docs_params.mappings:
                 multimodal_mappings: Dict[str, Dict] = utils.extract_multimodal_mappings(add_docs_params.mappings)
+
                 for field_name, multimodal_params in multimodal_mappings.items():
                     if not utils.is_tensor_field(field_name, add_docs_params.tensor_fields):
                         raise errors.InvalidArgError(f"Multimodal field {field_name} must be a tensor field")
 
+                    try:
+                        unstructured_vespa_index.validate_multimodal_field_name(field_name)
```

Comment:
Do this validation before any inference as there's no point if the doc is invalid